### PR TITLE
api: add /api/debug/health

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: ref #161 

Problem Summary: Depends on #191 , a better API for health checking. Also fixes for dockerfile. Also add `tiproxyctl health`.

Well-tested with the new tidb-operator PR.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Notable changes

- [ ] Has configuration change
- [x] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [x] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by djshow832** [comment]
Please fix conflicts first

**Comment by xhebox** [comment]
> Please fix conflicts first

Done

File: lib/cli/health.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 PingCAP, Inc.
```
Comment (djshow832) [inline lib/cli/health.go:1]:
```suggestion
// Copyright 2023 PingCAP, Inc.
```

**Review by djshow832 — COMMENTED** [review]
Rest LGTM

**Comment by djshow832** [comment]
I didn't see any unit tests, please remove the `Unit Test` checkbox in the PR description.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**